app/views/ontology_versions/_ontology_version.html.haml

```diff
@@ -4,4 +4,14 @@
   %td= timestamp ontology_version
   %td= ontology_version.user ? fancy_link(ontology_version.user) : '-'
   %td= render partial: 'shared/state_tag', locals: {resource: ontology_version}
-  %td= link_to 'Download', fancy_repository_path(resource_chain[0], path: ontology_version.path, ref: ontology_version.commit_oid, action: :download), class: 'btn btn-xs btn-default'
+  %td
+    = link_to 'Download', fancy_repository_path(resource_chain[0], path: ontology_version.path, ref: ontology_version.commit_oid, action: :download), class: 'btn btn-xs btn-default'
+    - latest_version = ontology_version_counter == 0
+    - if can?(:write, ontology.repository) && ontology_version.state == 'failed' && latest_version
+      - btn_text = 'Edit'
+      - anchor = '#edit'
+    - else
+      - btn_text = 'View'
+      - anchor = ''
+    - exact_commit = !latest_version
+    = link_to btn_text, fancy_repository_path(resource_chain[0], path: ontology_version.path, ref: ontology_version.commit_oid, action: :show, exact_commit: exact_commit) + anchor, class: 'btn btn-xs btn-default'
```

Comment:
Please put the business logic (assigning the variables) of the view into a helper method to keep the view slim and stupid. You could, for instance, put all the code in the `td` except for the download button into a method `edit_or_view_button`.